pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -493,19 +494,29 @@ impl<T: Config> Module<T> {
     }
 
     fn _bill_contract(contract: &mut types::Contract) -> DispatchResult {
+        let now = <timestamp::Module<T>>::get().saturated_into::<u64>() / 1000;
+        let mut contract_billing_info = ContractBillingInformationByID::get(contract.contract_id);
+
+        match contract.contract_type {
+            types::ContractData::NodeContract(_) => {
+                if contract_billing_info.amount_unbilled == 0 {
+                    ContractLastBilledAt::insert(contract.contract_id, now);
+                    return Ok(())
+                }
+            },
+            _ => ()
```

Comment:
Considering you only want to know _if_ it is a node contract here, and not do anything with the data, a `matches!(contract.contract_type, types::ContractData::NodeContract(_)) { ... }` would be cleaner here

Reply:
thanks :)